### PR TITLE
Christy - make WBS names active links

### PR DESCRIPTION
**Pull request by chappdev2019**
# Description
image

This PR converts WBS names into active links within the Projects section of the Admin user's Reports page. Besides, WBS names are displayed as active links only for users with permissions, enabling direct navigation to the WBS details.
…

## Related PRS (if any):
None.
…

## Main changes explained:
- Update file WbsTable.jsx: Converts WBS names into active, clickable links within the Projects section of the Admin user's Reports page.
- Update file ProjectReport.jsx: Enforces permission checks so that only users with the appropriate permissions can click the WBS links. Users without permission will not see the links.
…

## How to test:
1. Check into current branch.
2. Do npm install and npm run start:local to run this PR locally.
3. Clear site data/cache.
4. Log in with an Admin account.
5. Go to dashboard→ Reports→ Reports→ Projects→ Choose one project → WBS names should be active links here.
6. Log in with a user account that has been granted below any of the listed permissions related to Work Breakdown Structures.

image

7. Go to dashboard→ Reports→ Reports→ Projects→ Choose one project → WBS names should be active links here.
8. Log in with a user account that has no permission related to Work Breakdown Structures.
9. Go to dashboard→ Reports→ Reports→ Projects→ Choose one project → WBS names will not be links.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59427068/13fcec3a-d642-401e-a037-9509c106bf99

## Note:
This permission must be enabled to test the PR.
image

### CODE REVIEW

**Review by meetpadhiar4 — APPROVED** [review]
Hi, I reviewed your PR and all the changes are working correctly. I tested this feature with admin account and volunteer account with and without permissions. Please refer the videos attached.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/5edaa1fa-3436-44ad-b11f-6f36f35821da


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/9bc7eb6d-6c96-4c2e-81c1-0a5e18ed4ff0


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/e047ef8e-d490-4353-97ae-b7ba02188dc6

**Review by swathyjayaseelan — APPROVED** [review]
The changes look good. I tested the changes with and without permission and both works. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13806470/6253db3d-5685-4a13-bd69-c3d9e263858e

**Review by XiaohanMeng — APPROVED** [review]
Hi, I tested this PR, and everything worked as intended. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/0ce4a15b-8a31-4b15-b079-dca40a38da94

**Review by psharma1984 — APPROVED** [review]
Hi @mianmiantea2019 The code works as expected. I think the WBS button link has changed therefore i see it differently for my volunteer account. I have tested with and without permission. Please see below for reference

[screen-capture (2).webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123996264/809e5484-232a-4133-a000-8baa6be5bddc)

[screen-capture (3).webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123996264/d80df859-38dc-4568-9743-996c99a338d4)

**Review by skaush21 — APPROVED** [review]
Features match description
* logged in with owner account, navigated to reports -> reports -> project -> wbs, names are linked
* assigned permissions to a volunteer account, logged out
* signed in to volunteer account, navigated to reports -> reports -> project -> wbs
* verified that names are still linked
* logged off and sign in to a new volunteer account, reports is no longer visible
Features implemented work

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/141872402/7a062c39-7d68-4071-9864-a15dae023f98

**Review by ClemarNunes — APPROVED** [review]
I tested your PR and worked as expected.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/ff708884-eedd-42c2-9cf3-ada2afbcf47e

**Review by bhuvan-dv — APPROVED** [review]
The changes look good, works as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67942070/7be9ce53-b6d3-4d30-a669-6717d7036d39

**Review by FyisFe — APPROVED** [review]
I tested your PR with users with and without permission. Everything works as expected.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59299011/3ceb8933-b198-40a3-9598-09303a7a24ac

**Comment by hetvi11110** [comment]
I tested the code with user permissions for Work Breakdown Structures as an admin role and another without permissions for Work Breakdown Structures as a core team role. Both are working as described. Here is the reference video.
[screencast-chat.openai.com-2024.03.01-20_37_30.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13557067/ad1fccb0-18eb-44e3-a3e0-3b8c00dafa66)

**Comment by hetvi11110** [comment]
I tested the code with user permissions for Work Breakdown Structures as an admin role and another without permissions for Work Breakdown Structures as a core team role. Both are working as described. Here is the reference 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13557067/59a238dc-c3d3-4abf-a6f0-f5617315fa2e

**Review by sucheta90 — APPROVED** [review]
Hello @mianmiantea2019 , I have tested this PR and it worked as expected. At first, I logged in as an Admin user, checked the Reports tab, and then upon clicking on the Project - tile I was able to select between lists of projects. The ones that had a WBS link would take me to the related task/ tasks.  Also, I was able to grant permission to have access to Reports for my test volunteer account and it worked as expected. Well done.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/101291283/f9e3f9da-f0f6-4d1b-9b0d-00278554bc54

**Review by PratimaSingh02 — APPROVED** [review]
I tested this PR with an Admin and a Volunteer account. It worked as described.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/48998232/de0f40ec-4ec1-4f87-94b1-4802d35533a2


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/48998232/57e202b0-9b3c-409a-b68c-199da8cd7289

**Review by DiegoSalas27 — CHANGES_REQUESTED** [review]
Hi Christy. I spotted an error when a user without permissions to see the wbs structure components can actually see them if they copy and paste the non-protected url in their browser.

In short, the only thing left is to make those wbs related routes protected. I have attached two videos (the first one tests the funcitonality, and showcases this flaw; the second one will help you implementing the protected route). 

Please let me know if there is anything I can help you with. Overall great work.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26840798/edd341d7-b4cf-4892-a21d-b014d801da64


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26840798/6ef36e83-cb66-4cbf-9230-9fb170a1c44e

**Review by MALAV1308 — APPROVED** [review]
Hello, I reviewed your PR and it is working perfectly. All three test cases passed. Great Job!
user-1
user

**Review by huangweihou — APPROVED** [review]
I tested admin and worked correctly. For the volunteer account with and without permissions as well. Please take the below screenshot as reference:

**Review by hetvi11110 — APPROVED** [review]

**Review by navyasrim — APPROVED** [review]
Tested the PR and on clicking it is navigating to the WBS..as mentioned checked with volunteer and admin..
PR-1985

**Review by Sarthak32 — APPROVED** [review]
I tested for both the roles. It works correctly.
Screenshot 2024-03-03 112046
Screenshot 2024-03-03 112812

**Review by Gugor99 — APPROVED** [review]
### Report
Hi Christy,
I tried your implementation and it worked as expected. As suggested, I logged in with two different accounts both with permission to see the listed WBS links, and I was indeed able to click and check the links. I then logged with a account that had no permission and I wasn't able to access the links, I didn't notice any bug so well done!
### Screenshots
Cattura1
Cattura2

**Comment by Weiyao-Li** [comment]
Admin - Active Link Checked: 
image


User part:
image

**Comment by mianmiantea2019** [comment]
> Hi Christy. I spotted an error when a user without permissions to see the wbs structure components can actually see them if they copy and paste the non-protected url in their browser.
> 
> In short, the only thing left is to make those wbs related routes protected. I have attached two videos (the first one tests the funcitonality, and showcases this flaw; the second one will help you implementing the protected route).
> 
> Please let me know if there is anything I can help you with. Overall great work.
> 
>  screen-recording-2024-03-02-at-113019-am_3ebFT1u4.mp4 
>  screen-recording-2024-03-02-at-115451-am_I3EZlgcp.mp4

Hi Diego! That's a valid point. This pull request primarily aims to make the WBS name an active link. So, let me create a new requirement to address the protected route. Thank you for your demo, and I'll definitely reach out if I need anything else!

**Review by Coops023 — APPROVED** [review]
Hi Christy, i have reviewed this PR, i logged in as an admin, and was able to see WBS as a link in the project section. I then logged in as another user without this permission and the link was not active. well done. 

PR1985

PR1985-2

**Review by Tiantian-C — COMMENTED** [review]
I have tested it with an admin account with full permission and another account without permission and it looks nice!!
Screenshot 2024-03-06 at 4 39 28 PM
Screenshot 2024-03-06 at 4 38 55 PM

**Comment by one-community** [comment]
Thank you all, moving to final review.  Cristy, please address Diego's request and get his approval after.

**Review by cvtqx — APPROVED** [review]
Works as described, great job!

Testing as admin with permissions, WBS are active links:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/12f2a370-7711-4aff-9f9b-47e8a0cb2d8f

Testing as a user without permissions, WBS are not links:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/107726913/047a481b-4a1e-459f-8388-54eb6d1ae3e5

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**
